ironic/tests/unit/drivers/modules/drac/test_management.py

```diff
@@ -445,6 +466,100 @@ def test_set_boot_device_with_invalid_job_queue(
+
+        drac_mgmt.set_boot_device(
+            self.node, ironic.common.boot_devices.PXE, persistent=False)
+        mock_validate_job_queue.assert_called_once_with(self.node)
+
+    @mock.patch.object(drac_job, 'validate_job_queue', spec_set=True,
+                       autospec=True)
+    @mock.patch.object(drac_job, 'list_unfinished_jobs', spec_set=True,
+                       autospec=True)
+    def test_set_boot_device_with_list_unfinished_jobs_fail(
+            self, mock_list_unfinished_jobs, mock_validate_job_queue,
+            mock_get_drac_client):
+        mock_client = mock.Mock()
+        mock_get_drac_client.return_value = mock_client
+
+        mock_list_unfinished_jobs.side_effect = exception.DracOperationError(
+            'boom')
+
+        self.assertRaises(exception.DracOperationError,
+                          drac_mgmt.set_boot_device, self.node,
+                          ironic.common.boot_devices.PXE, persistent=True)
+
+        self.assertEqual(0, mock_client.change_boot_device_order.call_count)
+        self.assertEqual(0, mock_client.set_bios_settings.call_count)
+        self.assertEqual(0, mock_client.commit_pending_bios_changes.call_count)
+
+    @mock.patch.object(drac_job, 'list_unfinished_jobs', spec_set=True,
+                       autospec=True)
+    @mock.patch.object(drac_job, 'validate_job_queue', spec_set=True,
+                       autospec=True)
+    @mock.patch.object(drac_mgmt, '_get_boot_device', spec_set=True,
+                       autospec=True)
+    @mock.patch.object(drac_mgmt, '_get_next_persistent_boot_mode',
+                       spec_set=True, autospec=True)
+    def test_set_boot_device_with_list_unfinished_jobs(
+            self, mock__get_next_persistent_boot_mode, mock__get_boot_device,
+            mock_validate_job_queue, mock_list_unfinished_jobs,
+            mock_get_drac_client):
+        mock_client = mock.Mock()
+        mock_get_drac_client.return_value = mock_client
+
+        bios_job_dict = {
+            'id': 'JID_602553293345',
+            'name': 'ConfigBIOS:BIOS.Setup.1-1',
+            'start_time': 'TIME_NOW',
+            'until_time': 'TIME_NA',
+            'message': 'Task successfully scheduled.',
+            'status': 'Scheduled',
+            'percent_complete': 0}
+
+        job = test_utils.make_job(bios_job_dict)
+        mock_list_unfinished_jobs.return_value = [job]
+        mock_client.list_boot_devices.return_value = self.boot_devices['IPL']
+        boot_device = {'boot_device': ironic.common.boot_devices.DISK,
+                       'persistent': True}
+
+        mock__get_boot_device.return_value = boot_device
+        mock__get_next_persistent_boot_mode.return_value = 'IPL'
+
+        drac_mgmt.set_boot_device(self.node, ironic.common.boot_devices.DISK,
+                                  persistent=True)
+        mock_client.delete_jobs.assert_called_once_with(
+            job_ids=['JID_602553293345'])
+
```

Comment:
Let's add one more test case that tests list_unfinished jobs returning a bios job and another job

Reply:
Thanks, will add it.

Reply:
I have added test_set_boot_device_with_multiple_unfinished_jobs test case, please review this.